nilearn/glm/contrasts.py

```diff
@@ -262,6 +265,35 @@ def p_value(self, baseline=0.0):
         self.p_value_ = p_values
         return p_values
 
+    def one_minus_pvalue(self, baseline=0.0):
+        """Return a parametric estimate using
+        the cumulative distribution function of the p-value associated
+        with the null hypothesis: (H0) 'contrast equals baseline'
+
+        Parameters
+        ----------
+        baseline : float, optional
+            baseline value for the test statistic
+
+        Returns
+        -------
+        one_minus_pvalues : 1-d array, shape=(n_voxels,)
+            one_minus_pvalues, one per voxel
+        """
+        if self.stat_ is None or not self.baseline == baseline:
+            self.stat_ = self.stat(baseline)
```

Comment:
could you add a test for this case?

Reply:
This is basically a replication of the previous function `p_value` but using cdf instead of sf. I don't find a concrete test function for `p_value` that I could also replicate for this one. What do you suggest?

Reply:
I understand that this line has not been tested and that's why codecov/patch is complaining. I think I should add a test where also `p_value` function is tested but I don't find it.

Reply:
maybe just calling `p_value` and `one_minus_pvalue` for models that haven't computed `stat_` yet?

Reply:
I decided to remove this conditional statement as well as the one from the next comment. The reason these lines were not being tested by codecov/patch---and therefore producing an error---is because the exactly same conditional statements are always tested before in the function `p_value`, which takes place right before this one. Let me know whether you agree with this solution. @jeromedockes @bthirion

Reply:
I'm not sure why remove them? it is still possible for a user to specify an
invalid contrast type (different from "t" and "F"), and also to call
`one_minus_pvalue` without calling `stat` first -- so these checks are still
necessary. what was missing was a test covering these 2 situations.

Reply:
@alpinho  AFAICT this is still to be addressed? ie handle the cases where the contrast type is invalid, and where `stat_` doesn't exist, in `one_minus_pvalue`, and add corresponding tests. I think that's the only one left!

Reply:
Sorry, I didn't see this comment before. I think the reason codecov/patch was producing an error is because these lines I removed were never being tested because they were always tested before in the `p_value` function. So, I am not sure whether that's true that it is still possible for a user to specify an invalid contrast type (different from "t" and "F") and also call `one_minus_pvalue` without calling `p_value` before. But I am happy to solve this problem in a different way; I just don't know exactly how. Concretely, I don't know where to add the tests for this; it should be in the same place where the same lines of `p_value` function are tested, but I don't find them...

Reply:
for example:
```python
import numpy as np
from nilearn.glm.contrasts import Contrast

effect = np.ones((1, 3))
variance = effect[0]

contrast = Contrast(effect, variance, contrast_type="t")
# fails because stat() has not been called
print(contrast.one_minus_pvalue())

# fails with cryptic error message because bad stat type not handled
contrast = Contrast(effect, variance, contrast_type="bad")
print(contrast.one_minus_pvalue())
```

Reply:
I would assume it should be in `nilearn/glm/tests/test_contrasts.py` but, if that's the case, where exactly?

Reply:
you can create a test `test_one_minus_pvalue`

Reply:
maybe something like
```python
def test_one_minus_pvalue():
    effect = np.ones((1, 3))
    variance = effect[0]

    contrast = Contrast(effect, variance, contrast_type="t")
    assert np.allclose(contrast.one_minus_pvalue(), 0.84, 1)
    assert np.allclose(contrast.stat_, 1., 1)

    contrast = Contrast(effect, variance, contrast_type="bad")
    with pytest.raises(ValueError):
        contrast.one_minus_pvalue()

```

Reply:
Thanks! I was working on this when I received your last suggestion. I did the second test in a very similar way but using numpy.testing. I was trying to figure out how to do the first test but you gave me a great a help. I copied/pasted it. I just committed everything. Let me know. Thanks!

Reply:
There's still one segment in red in Codecov Report diagram...

Reply:
ah yes, because the exception already gets raised in `stat()` when the stat type is invalid. that's ok, don't worry about it

Reply:
OK.